fraud/sync.go

```diff
@@ -0,0 +1,84 @@
+package fraud
+
+import (
+	"context"
+
+	"github.com/libp2p/go-libp2p-core/event"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+
+	pb "github.com/celestiaorg/celestia-node/fraud/pb"
+)
+
+func (f *ProofService) syncFraudProofs(ctx context.Context) {
+	log.Debug("start fetching Fraud Proofs")
+	sub, err := f.host.EventBus().Subscribe(&event.EvtPeerIdentificationCompleted{})
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	defer sub.Close()
+	f.topicsLk.RLock()
```

Comment:
```suggestion
	f.topicsLk.RLock()
	topics := f.topics
	f.topicsLk.RUnlock()
	....etc
```

Reply:
f.topics is a map. map is a reference type in golang. It means that accessing `topics` without mutex lock will still cause  a data race. To avoid DR we have hard copy f.topics using `for k,v := range f.topics`. But in this case I do not see any reason for doing it